This patch is based on this fix from upstream: https://bugs.webkit.org/show_bug.cgi?id=171612

It fixes incopatibility in ICU 59.1. This patch can be removed once we update
to webkitgtk containing this fix.

--- webkitgtk-2.8.4/Source/JavaScriptCore/API/JSStringRef.cpp	2017-10-19 13:49:35.684251808 +0000
+++ webkitgtk-2.8.4/Source/JavaScriptCore/API/JSStringRef.cpp	2017-10-19 13:49:31.408054685 +0000
@@ -37,7 +37,7 @@ using namespace WTF::Unicode;
 JSStringRef JSStringCreateWithCharacters(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(chars, numChars).leakRef();
+    return OpaqueJSString::create(reinterpret_cast<const UChar*>(chars), numChars).leakRef();
 }
 
 JSStringRef JSStringCreateWithUTF8CString(const char* string)
@@ -62,7 +62,7 @@ JSStringRef JSStringCreateWithUTF8CStrin
 JSStringRef JSStringCreateWithCharactersNoCopy(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
+    return OpaqueJSString::create(StringImpl::createWithoutCopying(reinterpret_cast<const UChar*>(chars), numChars)).leakRef();
 }
 
 JSStringRef JSStringRetain(JSStringRef string)
@@ -87,7 +87,7 @@ const JSChar* JSStringGetCharactersPtr(J
 {
     if (!string)
         return nullptr;
-    return string->characters();
+    return reinterpret_cast<const JSChar*>(string->characters());
 }
 
 size_t JSStringGetMaximumUTF8CStringSize(JSStringRef string)
--- webkitgtk-2.8.4/Source/WebKit2/Shared/API/c/WKString.cpp	2017-10-19 16:26:11.057567274 +0000
+++ webkitgtk-2.8.4/Source/WebKit2/Shared/API/c/WKString.cpp	2017-10-19 16:25:36.772074257 +0000
@@ -58,7 +58,7 @@ size_t WKStringGetLength(WKStringRef str
 size_t WKStringGetCharacters(WKStringRef stringRef, WKChar* buffer, size_t bufferLength)
 {
     COMPILE_ASSERT(sizeof(WKChar) == sizeof(UChar), WKStringGetCharacters_sizeof_WKChar_matches_UChar);
-    return (toImpl(stringRef)->getCharacters(static_cast<UChar*>(buffer), bufferLength));
+    return (toImpl(stringRef)->getCharacters(reinterpret_cast<UChar*>(buffer), bufferLength));
 }
 
 size_t WKStringGetMaximumUTF8CStringSize(WKStringRef stringRef)
